src/Domain/CoordinateMaps/PolarToCartesian.cpp

```diff
@@ -0,0 +1,100 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/CoordinateMaps/PolarToCartesian.hpp"
+
+#include <cmath>
+#include <cstddef>
+#include <pup.h>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/DereferenceWrapper.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+namespace domain::CoordinateMaps {
+PolarToCartesian::PolarToCartesian() = default;
+PolarToCartesian::PolarToCartesian(PolarToCartesian&&) = default;
+PolarToCartesian::PolarToCartesian(const PolarToCartesian&) = default;
+PolarToCartesian& PolarToCartesian::operator=(const PolarToCartesian&) =
+    default;
+PolarToCartesian& PolarToCartesian::operator=(PolarToCartesian&&) = default;
+
+template <typename T>
+std::array<tt::remove_cvref_wrap_t<T>, 2> PolarToCartesian::operator()(
+    const std::array<T, 2>& source_coords) const {
+  const auto& [r, phi] = source_coords;
+  return {{r * cos(phi), r * sin(phi)}};
+}
+
+// NOLINTNEXTLINE(readability-convert-member-functions-to-static)
+std::optional<std::array<double, 2>> PolarToCartesian::inverse(
+    const std::array<double, 2>& target_coords) const {
+  const auto& [x, y] = target_coords;
+  if (UNLIKELY(y == 0.0 and x == 0.0)) {
+    return std::array{0.0, 0.0};
+  } else {
+    const double r = std::hypot(x, y);
+    return std::array{r, y < 0.0 ? atan2(y, x) + 2.0 * M_PI : atan2(y, x)};
```

Comment:
Issues when `x < 0` and `y` is negative 0.  Instead, do
```c++
const double phi = atan2(y, x);
return std::array{r, phi < 0.0 ? phi + 2.0 * M_PI : phi};
```
For the test:
```c++
test_map_at_point(map, {{1.0, M_PI}}, {{-1.0, std::copysign(0.0, -1.0)}});
```
(There are more natural ways to get negative zero, but for a test best to be obvious.)

Same for the pfaffian version.